pkg/ring/kv/etcd/etcd.go

```diff
@@ -34,10 +42,39 @@ func (cfg *Config) RegisterFlagsWithPrefix(f *flag.FlagSet, prefix string) {
 	f.Var((*flagext.StringSlice)(&cfg.Endpoints), prefix+"etcd.endpoints", "The etcd endpoints to connect to.")
 	f.DurationVar(&cfg.DialTimeout, prefix+"etcd.dial-timeout", 10*time.Second, "The dial timeout for the etcd connection.")
 	f.IntVar(&cfg.MaxRetries, prefix+"etcd.max-retries", 10, "The maximum number of retries to do for failed ops.")
+	f.BoolVar(&cfg.EnableTLS, prefix+"etcd.tls-enabled", false, "Enable TLS")
+	f.StringVar(&cfg.CertFile, prefix+"etcd.tls-cert-file", "", "The TLS certificate file path")
+	f.StringVar(&cfg.KeyFile, prefix+"etcd.tls-key-file", "", "The TLS private key file path")
+	f.StringVar(&cfg.TrustedCAFile, prefix+"etcd.tls-client-ca-file", "", "The trusted CA file path")
+	f.BoolVar(&cfg.InsecureSkipVerify, prefix+"etcd.tls-insecure-skip-verify", false, "Skip validating server certificate")
+}
+
+// GetTLS sets the TLS config field with certs
+func (cfg *Config) GetTLS() (*tls.Config, error) {
+	if cfg.CertFile == "" && cfg.KeyFile == "" && !cfg.EnableTLS {
+		return nil, nil
+	}
+	tlsInfo := &transport.TLSInfo{
+		CertFile: cfg.CertFile,
+		KeyFile:  cfg.KeyFile,
+	}
+	if cfg.TrustedCAFile != "" {
+		tlsInfo.TrustedCAFile = cfg.TrustedCAFile
```

Comment:
No need for the `if` because the empty string is also the zero value of the string in the `transport.TLSInfo` structure.

```suggestion
		tlsInfo.TrustedCAFile = cfg.TrustedCAFile
```